capnp/src/text.rs

```diff
@@ -44,7 +44,7 @@ pub type Reader<'a> = &'a str;
 pub fn new_reader(v: &[u8]) -> Result<Reader<'_>> {
     match str::from_utf8(v) {
         Ok(v) => Ok(v),
-        Err(e) => Err(Error::failed(format!("Text contains non-utf8 data: {e:?}"))),
+        Err(_e) => Err(Error::from_kind(ErrorKind::TextContainsNonUtf8Data)),
```

Comment:
Can we make the `ErrorKind` variant carry along the `core::str::Utf8Error` here?